ffi/src/engine_funcs.rs

```diff
@@ -103,29 +96,36 @@ pub unsafe extern "C" fn free_read_result_iter(data: Handle<MutFileReadResultIte
 /// # Safety
 /// Caller is responsible for calling with a valid `ExternEngineHandle` and `FileMeta`
 #[no_mangle]
-pub unsafe extern "C" fn read_parquet_files(
+pub unsafe extern "C" fn read_parquet_file(
     engine: Handle<SharedExternEngine>,
     file: &FileMeta,
     physical_schema: Handle<SharedSchema>,
-) -> ExternResult<Handle<MutFileReadResultIterator>> {
-    let extern_engine = unsafe { engine.clone_as_arc() };
+) -> ExternResult<Handle<ExclusiveFileReadResultIterator>> {
+    let engine = unsafe { engine.clone_as_arc() };
     let physical_schema = unsafe { physical_schema.clone_as_arc() };
-    let res = read_parquet_files_impl(extern_engine.clone(), file, physical_schema);
-    res.into_extern_result(&extern_engine.as_ref())
+    let path = unsafe { String::try_from_slice(&file.path) };
```

Comment:
aside: why is it called `try_from_slice` if it doesn't return a `Result`? Should we call it `from_slice` instead?

Reply:
It _does_ return a result. I just don't unwrap the result here. Note that `read_parquet_file_impl` takes a `DeltaResult<String>` and it uses `?` on that arg. That let's us have all the "error" cases handled by the single return from `read_parquet_file_impl` two lines below.